plugins/payment-plugin/payment_plugin.yaml

```diff
@@ -0,0 +1,20 @@
+name: PaymentPlugin
+description: |
+    The name of the plugin is payment. This plugin takes payment amount to make a payment for a specific reason.
+
+CREDENTIALS:
```

Comment:
Why did you change the YAML?

Reply:
To make it understandable by codegen.

Reply:
NL2DSL doesn't understand credentials

Reply:
It doesn't matter. Credentials will be passed by codegen during plugin initialization. 
Here is an example of plugin initialization with credentials, in the calling FSM:

```python
self.plugins: Dict[str, AbstractFSM] = {
     "payment": PaymentPlugin(
            send_message=send_message,
            credentials={
                    "RAZORPAY_CLIENT_ID": self.credentials.get("RAZORPAY_CLIENT_ID"),
                    "RAZORPAY_CLIENT_SECRET": self.credentials.get("RAZORPAY_CLIENT_SECRET"),
            },
     ),
}
```

Codegen can handle this initialisation.

Reply:
But not NL2DSL. We will solve this later. I don't want to disturb NL2DSL. At the moment take all arguments through the function call and not through constructor.

Reply:
NL2DSL doesn't have to change to incorporate credentials. Since credentials names are already defined in this yaml description, codegen can generate the above code snippet. We don't need to change NL2DSL.